src/KeyedFrames.jl

```diff
@@ -106,13 +109,13 @@ Base.getindex(kf::KeyedFrame, ::Colon) = copy(kf)
 Base.getindex(kf::KeyedFrame, ::Colon, ::Colon) = copy(kf)
 
 # Returns a KeyedFrame
-Base.getindex(kf::KeyedFrame, col::AbstractVector) = _kf_getindex(kf, col)
+Base.getindex(kf::KeyedFrame, ::typeof(!), col::AbstractVector) = _kf_getindex(kf, !, col)
 
 # Returns a column
-Base.getindex(kf::KeyedFrame, col::ColumnIndex) = frame(kf)[col]
+Base.getindex(kf::KeyedFrame, ::typeof(!), col::ColumnIndex) = frame(kf)[!, col]
 
 # Returns a KeyedFrame or a column (depending on the type of col)
-Base.getindex(kf::KeyedFrame, ::Colon, col) = kf[col]
+Base.getindex(kf::KeyedFrame, ::Colon, col) = kf[!, col]
```

Comment:
```suggestion
Base.getindex(kf::KeyedFrame, ::Colon, col) = kf[:, col]
```

Reply:
This doesn't work, is just recursive, the `!` behaviour is what was happening before with `kf[col]`

Reply:
DataFrames promises a defensive copy when you index with `:` rather than `!`
So either:
```suggestion
Base.getindex(kf::KeyedFrame, ::Colon, col) = copy(kf[!, col])
```
or
```suggestion
Base.getindex(kf::KeyedFrame, ::Colon, col) = kf.frame[:, col]
```